dev/ProgressRing/ProgressRing.cpp

```diff
@@ -147,32 +228,54 @@ void ProgressRing::SetLottieBackgroundColor(winrt::impl::com_ref<AnimatedVisuals
         }
         else
         {
-            // Default color fallback if Background() Brush does not contain SolidColorBrush with Color property.
+             //Default color fallback if Background() Brush does not contain SolidColorBrush with Color property.
             return SharedHelpers::FindInApplicationResources(s_DefaultBackgroundThemeResourceName).as<winrt::SolidColorBrush>().Color();
         }
     }();
 
-    progressRingIndeterminate->GetThemeProperties(compositor).InsertVector4(s_BackgroundName, SharedHelpers::RgbaColor(backgroundColor));
+    if (const auto progressRing = animatedVisualSource.try_as<AnimatedVisuals::ProgressRingIndeterminate>())
+    {
+        progressRing->GetThemeProperties(compositor).InsertVector4(s_BackgroundName, SharedHelpers::RgbaColor(backgroundColor));
+    }
 }
 
-void ProgressRing::ChangeVisualState()
+void ProgressRing::UpdateLottieProgress()
 {
-    if (IsActive())
+    if (auto&& determinatePlayer = m_determinatePlayer.get())
     {
-        winrt::VisualStateManager::GoToState(*this, L"Active", true);
+        const double range = Maximum() - Minimum();
+        const double fromProgress = m_oldValue / range;
+        const double toProgress = Value() / range;
+
+        determinatePlayer.PlayAsync(fromProgress, toProgress, false);
+        m_oldValue = Value();
+    }
+}
 
-        if (auto&& player = m_player.get())
+void ProgressRing::UpdateStates()
+{
+    if (IsActive() && IsIndeterminate())
+    {
+        winrt::VisualStateManager::GoToState(*this, L"IndeterminateActive", true);
+
+        if (auto&& indeterminatePlayer = m_indeterminatePlayer.get())
         {
-            const auto _ = player.PlayAsync(0, 1, true);
+            const auto _ = indeterminatePlayer.PlayAsync(0, 1, true);
```

Comment:
>indeterminatePlayer [](start = 27, length = 19)

why do you need to name this return parameters but not the one you get from the playasync call in UpdateLottieProgress?

Reply:
Thanks for catching that. I do need to do that for the one in UpdateLottieProgress, or the Helix tests will flag this with the error "Avoid unnamed objects with custom construction and destruction".